### PR TITLE
fix: curlimages CVE fix

### PR DESCRIPTION
**Pull request by SandhyaRavi2403**
**What problem does this PR solve?**:
patch image forcurlimages to fix CVE

`sandhya.ravi@GT9X7CVF5F kommander-applications % trivy image ghcr.io/mesosphere/dkp-container-images/docker.io/curlimages/curl:7.88.1-d2iq.0 --vuln-type os --ignore-unfixed | grep Total
2024-07-10T18:30:23+05:30	INFO	Vulnerability scanning is enabled
2024-07-10T18:30:23+05:30	INFO	Secret scanning is enabled
2024-07-10T18:30:23+05:30	INFO	If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2024-07-10T18:30:23+05:30	INFO	Please see also https://aquasecurity.github.io/trivy/v0.53/docs/scanner/secret#recommendation for faster secret detection
2024-07-10T18:30:24+05:30	INFO	Detected OS	family="alpine" version="3.16.2"
2024-07-10T18:30:24+05:30	INFO	[alpine] Detecting vulnerabilities...	os_version="3.16" repository="3.16" pkg_num=21
2024-07-10T18:30:24+05:30	WARN	This OS version is no longer supported by the distribution	family="alpine" version="3.16.2"
2024-07-10T18:30:24+05:30	WARN	The vulnerability detection may be insufficient because security updates are not provided
Total: 0 (UNKNOWN: 0, LOW: 0, MEDIUM: 0, HIGH: 0, CRITICAL: 0)
sandhya.ravi@GT9X7CVF5F kommander-applications % 
`

workflow link:
https://github.com/mesosphere/dkp-container-images/actions/runs/9874011145

**Which issue(s) does this PR fix?**:

https://jira.nutanix.com/browse/NCN-101394

**Special notes for your reviewer**:

**Does this PR introduce a user-facing change?**:

```release-note

```

**Checklist**

- [ ] If the PR adds a version bump, ensure there is no breaking change in Licensing model (or NA).
- [ ] If a chart is changed or app configuration is significantly changed, the chart version is correctly incremented (so that apps are not automatically upgraded from a previous version of DKP).

### CODE REVIEW

**Comment by mhrabovcin** [comment]
@SandhyaRavi2403 we will need to change the `/services/gatekeeper/3.16.3` version.

**Comment by SandhyaRavi2403** [comment]
> @SandhyaRavi2403 we will need to change the `/services/gatekeeper/3.16.3` version.

Updated the service name version

File: services/gatekeeper/3.16.4/defaults/cm.yaml [hunk 1]
```diff
@@ -25,8 +25,8 @@ data:
       probeWebhook:
         priorityClassName: system-cluster-critical
         image:
-          repository: curlimages/curl
-          tag: 7.88.1
```
Comment (mhrabovcin) [inline services/gatekeeper/3.16.4/defaults/cm.yaml:29]: Would it be possible to use a newer image from docker repository directly - https://hub.docker.com/r/curlimages/curl

Comment (SandhyaRavi2403) [reply services/gatekeeper/3.16.4/defaults/cm.yaml]: updated the image to **8.8.0**
`sandhya.ravi@GT9X7CVF5F kommander-applications % trivy image docker.io/curlimages/curl:8.8.0 --vuln-type os --ignore-unfixed | grep Total
2024-07-15T15:47:33+05:30	INFO	Vulnerability scanning is enabled
2024-07-15T15:47:33+05:30	INFO	Secret scanning is enabled
2024-07-15T15:47:33+05:30	INFO	If your scanning is slow, please try '--scanners vuln' to disable secret scanning
2024-07-15T15:47:33+05:30	INFO	Please see also https://aquasecurity.github.io/trivy/v0.53/docs/scanner/secret#recommendation for faster secret detection
2024-07-15T15:47:38+05:30	INFO	Detected OS	family="alpine" version="3.19.1"
2024-07-15T15:47:38+05:30	INFO	[alpine] Detecting vulnerabilities...	os_version="3.19" repository="3.19" pkg_num=32
2024-07-15T15:47:38+05:30	WARN	Using severities from other vendors for some vulnerabilities. Read https://aquasecurity.github.io/trivy/v0.53/docs/scanner/vulnerability#severity-selection for details.
Total: 10 (UNKNOWN: 0, LOW: 0, MEDIUM: 10, HIGH: 0, CRITICAL: 0)`

**Review by mhrabovcin — APPROVED** [review]

File: services/gatekeeper/3.16.4/defaults/cm.yaml (hunk 1 above)
Comment (mhrabovcin) [reply services/gatekeeper/3.16.4/defaults/cm.yaml]: Thank you!

**Pull request merged**